sycl/include/CL/sycl/detail/image_impl.hpp

```diff
@@ -283,7 +283,7 @@ class image_impl final : public SYCLMemObjT<AllocatorT> {
                     RT::PiEvent &OutEventToWait) override {
     void *UserPtr = InitFromUserData ? BaseT::getUserPtr() : nullptr;
 
-    RT::PiMemImageDesc Desc = getImageDesc();
+    RT::PiMemImageDesc Desc = getImageDesc(static_cast<bool>(UserPtr));
```

Comment:
Unclear... What about:
```suggestion
    RT::PiMemImageDesc Desc = getImageDesc(UserPtr != nullptr);
```
to avoid a cast?

Reply:
Done.